react-native-scripts/src/scripts/eject.js

```diff
@@ -96,28 +116,33 @@ Ejecting is permanent! Please be careful with your selection.
         newDisplayName = expName;
       }
 
-      log("We have a couple of questions to ask you about how you'd like to name your app:");
-      const { enteredName, enteredDisplayname } = await inquirer.prompt([
-        {
-          name: 'enteredDisplayname',
-          message: "What should your app appear as on a user's home screen?",
-          default: newDisplayName,
-          validate: s => {
-            return s.length > 0;
+      if(isInteractive) {
+        log("We have a couple of questions to ask you about how you'd like to name your app:");
+        const { enteredName, enteredDisplayname } = await inquirer.prompt([
+          {
+            name: 'enteredDisplayname',
+            message: "What should your app appear as on a user's home screen?",
+            default: newDisplayName,
+            validate: s => {
+              return s.length > 0;
+            },
           },
-        },
-        {
-          name: 'enteredName',
-          message: 'What should your Android Studio and Xcode projects be called?',
-          default: newName,
-          validate: s => {
-            return s.length > 0 && s.indexOf('-') === -1 && s.indexOf(' ') === -1;
+          {
+            name: 'enteredName',
+            message: 'What should your Android Studio and Xcode projects be called?',
+            default: newName,
+            validate: s => {
```

Comment:
While valid, I think this is unrelated to this PR.  I think your fix can come in as a separate PR.
(Thanks for the review)